Shared/Samples/Display dimensions/DisplayDimensionsView.swift

```diff
@@ -0,0 +1,105 @@
+// Copyright 2023 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct DisplayDimensionsView: View {
+    /// A map with no specified style.
+    @State private var map = Map()
+    
+    /// The dimensional layer added to the map.
+    @State private var dimensionLayer: DimensionLayer?
+    
+    /// A Boolean that indicates whether to show an error alert.
+    @State private var isShowingErrorAlert = false
+    
+    /// The error shown in the error alert.
+    @State private var error: Error? {
+        didSet { isShowingErrorAlert = error != nil }
+    }
+    
+    var body: some View {
+        MapView(map: map)
+            .task {
+                do {
+                    // Load the local mobile map package using a URL.
+                    let mapPackage = MobileMapPackage(fileURL: .edinburghPylonDimensions)
+                    try await mapPackage.load()
+                    
+                    // Set the map to the first map in the mobile map package.
+                    if let map = mapPackage.maps.first {
+                        // Set the minScale to maintain dimension readability.
+                        map.minScale = 4e4
+                        self.map = map
+                    } else {
+                        throw SetupError.noMap
+                    }
+                    
+                    // Set the dimension layer using the one on the map.
+                    if let layer = map.operationalLayers.first(where: { $0 is DimensionLayer }) as? DimensionLayer {
+                        dimensionLayer = layer
+                    } else {
+                        throw SetupError.noDimensionLayer
+                    }
+                } catch {
+                    self.error = error
+                }
```

Comment:
If you wanted to clean up the view code some, you could move all this or just the section between the do/catch into an async function on the view.